src/subscript/fmu_copy_revision/fmu_copy_revision.py

```diff
@@ -0,0 +1,656 @@
+    parser.add_argument(
+        "--all, -all, -a", action="store_true", dest="all", help="List all folders"
+    )
+    parser.add_argument(
+        "--verbosity, -v",
+        action="store_true",
+        dest="verbosity",
+        help="Enable logging (messages) for debugging",
+    )
+    parser.add_argument(
+        "--cleanup",
+        action="store_true",
+        dest="cleanup",
+        help="Remove (cleanup) any existing target if present, default is False",
+    )
+    parser.add_argument(
+        "--skipestimate, --skip",
+        "-s",
+        action="store_true",
+        dest="skipestimate",
+        help="If present, skip estimation of current revision size.",
+    )
+    parser.add_argument("--source", dest="source", type=str, help="Add source folder")
+    parser.add_argument("--target", dest="target", type=str, help="Add target folder")
+    parser.add_argument(
+        "--option",
+        dest="option",
+        type=str,
+        default="3",
+        help="Option for copy profile to use, default is 3",
+    )
+    parser.add_argument(
+        "--threads",
+        dest="threads",
+        type=str,
+        default="99",  # 99 for automatic
+        help="Number of threads, default is computed automatically",
+    )
+
+    parser.add_argument(
+        "--version",
+        action="version",
+        version="%(prog)s (subscript version " + __version__ + ")",
+    )
+
+    logger.info("Parsing commandline")
+    return parser
+
+
+class CopyFMU:
+    """Class for copying a FMU revision."""
+
+    def __init__(self):
+        """Instantiate object."""
+        self.args = None
+        self.folders = []
+        self.source = None
+        self.default_target = None
+        self.target = None
+        self.nthreads = None
+        self.option = 3
+        self.filter = ""
+        self.dirfilter = ""
+        self.batch = False
+        self.keepfolders = 0
+
+    def do_parse_args(self, args):
+        """Parse command line arguments."""
+
+        if args is None:
+            args = sys.argv[1:]
+
+        parser = get_parser()
+        self.args = parser.parse_args(args)
+
+    def check_folders(self):
+        """Check if potential fmu folders are present or list all if --all."""
+
+        current = pathlib.Path(".")
+        folders = [fil for fil in current.iterdir() if fil.is_dir()]
+        result = []
+        for folder in folders:
+            fname = folder.name
+            if not self.args.all:
+                if fname.startswith(("r", "1", "2", "3")):
+                    result.append(fname)
+            else:
+                result.append(fname)
+
+        if result:
+            result = sorted(result)
+
+        if not result:
+            print(
+                "No valid folders to list. Are you in the correct folder "
+                "above your revisions? Or consider --all option to list all folders."
+            )
+            sys.exit()
```

Comment:
Exception rather than a system exit?

Reply:
Will prefer sys.exit() since this is a end-point user script. There is no way the user could test on the Exception and do something else. A sys.exit() will give a clean exit while Exception will give a noisy message.